src/module/Entity/src/Entity/Controller/RepositoryController.php

```diff
@@ -46,11 +49,15 @@ public function addRevisionAction()
             $form->setData($this->getRequest()->getPost());
             if ($form->isValid()) {
                 $data = $form->getData();
-                $this->getRepositoryManager()->commitRevision($entity, $data);
+                $revision = $this->getRepositoryManager()->commitRevision($entity, $data);
+                if ($mayCheckout) {
+                    $this->getRepositoryManager()->checkoutRevision($entity, $revision);
+                    $successMessage = 'Your revision has been saved and is available';
+                } else {
+                    $successMessage = 'Your revision has been saved, it will be available once it get\'s approved';
+                }
                 $this->getEntityManager()->flush();
-                $this->flashMessenger()->addSuccessMessage(
-                    'Your revision has been saved, it will be available once it get\'s approved'
-                );
+                $this->flashMessenger()->addSuccessMessage($successMessage);
 
                 return $this->redirect()->toRoute('entity/repository/history', ['entity' => $entity->getId()]);
```

Comment:
I redirected to the history for consistency, since this is also done when checking out a revision. Maybe change this, too?